### ISSUE TITLE
Bug : Right click after selecting multiple files deselects selected files in compact and icon view

### ISSUE THREAD

**Issue by msegmx** [linuxmint/nemo#1261]
**What I want to do :** select files and right click to chose copy/remove/etc. command.
**What I do :** In compact, icon view or desktop I select multiple files via mouse and stop moving the mouse immediately after last file is marked "blue". Then I right click.
**What happens :** popup appears after all selected files getting "de-selected"

**Workaround :** after selecting files with mouse, moving the mouse a little bit on the last selected file and then right clicking does show right-click menu (popup) without the files getting de-selected.

(personal note: I'm good at Counter Strike, but on the long run the workaround becomes annoying.)

Is this a bug ?

(Nemo 3.0.6)

**Comment by xenopeek** [comment]
I think this is as expected. Right-click calls up the context menu for the item under the mouse pointer. If you're not hovering the pointer over one of the items in the selection you pop up the context menu for whatever is instead under the pointer (either the current directory or another file or directory in it).

**Comment by msegmx** [comment]
Thanks for the quick reply.

> Right-click calls up the context menu for the item under the mouse pointer.

Sorry, that's not true. Again, I'm selecting let's say 3 files and my mouse stops over the third file. I'm still on that file and a right click pops up a menu without any file-related commands.

I've recorded it and uploaded it to YouTube (45 sec) : [https://youtu.be/-fd3caXj9EU](url)

Please share your thoughts on it. Thanks.

**Comment by JosephMcc** [comment]
I think I can reproduce this but I find it hard to keep the mouse still enough to make it happen. It seems that some kind of focus change happens with even a tiny mouse movement and then it works as expected.

**Comment by msegmx** [comment]
Exactly. And as I said in first post on the long run it becomes really annoying (esp. for someone who comes from Windows). 

I find it hard to believe that the programmers of Nemo think of it as "expected behavior".
This is one of those tiny things that when fixed increases user experience a lot.

I hope it will be accepted as a bug and will be fixed in future releases.

Thanks and keep up the great work.

**Comment by JosephMcc** [comment]
> I find it hard to believe that the programmers of Nemo think of it as "expected behavior".

I don't think that is the case. I don't remember ever running into this but I probably don't have a steady enough hand :)

**Comment by msegmx** [comment]
Same here.. until I played Counter Strike :)

**Comment by Timmy93** [comment]
Linux Mint 17.3, Cinnamon 2.8.8, Nemo 2.8.7
The bug here is not present so it must be a regression with the new nemo version.

**Comment by artscoop** [comment]
I've been having the exact same bug since August. It is so annoying I was sure it would finally get fixed. But no. I've just tested something and it is even worse:
- Click and draw a selection box that ends above one file.
- Do not move.
- Click the left button. The file gets deselected (bug)
- Click again the left button. The file is not selected (bug)
- Proper behaviour is only restored after a mouse move.
It's as if the user was always clicking on an empty area of the view.

The bug has popped up for the first time somewhere between Nemo 3.0/3.2

**Comment by msegmx** [comment]
I'm on Nemo 3.2.2 right now and can reproduce the bugs described by @artscoop.

One more thing that should also get fixed :

* right-click a file (eg. File_A)
* File_A gets selected and a popup with proper commands shows up on top of it, so far so good
* now, while the popup over File_A is still visible, right-click another file (File_B)

expected behavior :
* popup over File_A disappears 
* File_B gets selected
* a new popup with proper commands appears on top of File_B

current behavior : 
* the visible popup over File_A disappears
* File_A DOES NOT get deselected (File_A is still blue)
* File_B DOES NOT get selected
* no new popup appears over File_B

current workaround : 
* right-clicking File_B a second time solves the problem

**Comment by ultraviolet-1986** [comment]
Can confirm this for Mint 18, 18.1, and LMDE (current as of writing). This issue is not present for me in 17.x series. Tested on a HP Pavilion G6 and Entroware Orion laptop.

Is there any way to trace the commit that broke this?

**Comment by cinnamoner** [comment]
The mechanics of the MSDE is no piece 'o cake. Obviously, it's the best desktop ever created.

**Comment by JosephMcc** [comment]
I'm not sure how many of these are even bugs in Nemo or just the way things work in Gtk and outside of our control. That would explain the difference between Mint 17 and Mint 18 behaviors. The version of Gtk is much newer in 18 and even in LMDE.

**Comment by ultraviolet-1986** [comment]
I guess that makes sense, I don't mind moving the touchpad back and forth to work around this but it's a little annoying. Shame there's no output when it happens to help pin it down. I can't imagine Nemo would have changed drastically enough between releases for this to be the problem now I come to think of it. Shucks, we may have all been barking up the wrong tree.

I'll try and investigate further, in any case, it'd be nice to know for its own sake. Thanks.

**Comment by kmkom** [comment]
Hello,
I just installed Debian Buster with Cinnamon 3.6.7 and in this version Nemo has the same file "deselect"-problem. Selecting files with the mouse "lasso" and then right click (without mouse movement) does immediately deselect the files :-(
In this Debian version Nautilus 3.26.3.1 is installed also and I tried to trigger the same problem there but was not able to achieve that. Nautilus never "deselect" the files, so Nautilus does not have this problem, only Nemo has.
If someone is familiar with the code base of the two programs, maybe they can find out what Nautilus does differently if one selects several files with the "lasso"?
